src/workerd/api/node/zlib-util.h

```diff
@@ -1,57 +1,261 @@
 // Copyright (c) 2017-2022 Cloudflare, Inc.
 // Licensed under the Apache 2.0 license found in the LICENSE file or at:
 //     https://opensource.org/licenses/Apache-2.0
-
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
 #include <workerd/jsg/jsg.h>
+#include <workerd/jsg/exception.h>
 
+#include "zlib.h"
 #include <kj/array.h>
 #include <kj/compat/brotli.h>
-#include "zlib.h"
+#include <kj/mutex.h>
+#include <kj/vector.h>
 
-#include <limits>
-#include <cstdint>
+#include <cstdlib>
 
 namespace workerd::api::node {
 
+#ifndef ZLIB_ERROR_CODES
+#define ZLIB_ERROR_CODES(V)      \
+  V(Z_OK)                        \
+  V(Z_STREAM_END)                \
+  V(Z_NEED_DICT)                 \
+  V(Z_ERRNO)                     \
+  V(Z_STREAM_ERROR)              \
+  V(Z_DATA_ERROR)                \
+  V(Z_MEM_ERROR)                 \
+  V(Z_BUF_ERROR)                 \
+  V(Z_VERSION_ERROR)             \
+
+inline const char* ZlibStrerror(int err) {
+#define V(code) if (err == code) return #code;
+  ZLIB_ERROR_CODES(V)
+#undef V
+  return "Z_UNKNOWN_ERROR";
+}
+#endif  // ZLIB_ERROR_CODES
+
+// Certain zlib constants are defined by Node.js itself
+static constexpr auto Z_MIN_CHUNK = 64;
+static constexpr auto Z_MAX_CHUNK = 128 * 1024;
```

Comment:
```suggestion
static constexpr auto Z_MAX_CHUNK = 128 * 1024 * 1024;
```

This is fixed in my PR too but just pointing it out here